awswrangler/neptune/_neptune.py

```diff
@@ -266,6 +267,77 @@ def to_rdf_graph(
     return client.write_sparql(query)
 
 
+BULK_LOAD_IN_PROGRESS_STATES = {"LOAD_IN_QUEUE", "LOAD_NOT_STARTED", "LOAD_IN_PROGRESS"}
+
+
+@_utils.check_optional_dependency(sparql, "SPARQLWrapper")
+def bulk_load(
+    client: NeptuneClient,
+    path: str,
+    iam_role: str,
+    df: Optional[pd.DataFrame] = None,
+    neptune_load_wait_polling_delay: float = 0.25,
+    load_parallelism: Literal["LOW", "MEDIUM", "HIGH", "OVERSUBSCRIBE"] = "HIGH",
+    s3_write_mode: Literal["overwrite", "append"] = "append",
+) -> None:
+    """
+    Write records into Amazon Neptune using the Neptune Bulk Loader.
+
+    The DataFrame will be written to S3 and then loaded to Neptune using the
+    `Bulk Loader <https://docs.aws.amazon.com/neptune/latest/userguide/bulk-load.html>`_.
+
+    If no DataFrame is provided, then the contents of the S3 path will be loaded onto
+    Neptune.
+
+    Parameters
+    ----------
+    client: NeptuneClient
+        Instance of the neptune client to use
+    path: str
+        S3 Path that the Neptune Bulk Loader will load data from.
+        If a Pandas DataFrame is provided, it will be written to this location.
+    iam_role: str
+        The Amazon Resource Name (ARN) for an IAM role to be assumed by the Neptune DB instance for access to the S3 bucket.
+            For information about creating a role that has access to Amazon S3 and then associating it with a Neptune cluster,
+            see `Prerequisites: IAM Role and Amazon S3 Access <https://docs.aws.amazon.com/neptune/latest/userguide/bulk-load-tutorial-IAM.html>`_.
+    df: DataFrame, optional
+        `Pandas DataFrame <https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.html>`_ to write to Neptune.
+        If none is provided, the data which already exists in the S3 path will be loaded.
+    neptune_load_wait_polling_delay: float
+        Interval in seconds for how often the function will check if the Neptune bulk load has completed.
+    load_parallelism: str
+        Specifies the number of threads used by Neptune's bulk load process.
+    s3_write_mode: str
+        Specifies whether the Pandas DataFrame will ovewrite the contents of the S3 path, or merely append to it.
+        The entire content of the S3 path will be loaded into Neptune regardless.
+        Only applicable when ``df`` is provided.
+
+    """
+    if df is not None:
+        s3.to_csv(df, path, dataset=True, mode=s3_write_mode, index=False)
```

Comment:
Shall we also clean path after the load / add `keep_files` flag like we do in other similar APIs? 

Also to be honest, since `s3.to_csv` can be easily composed with `neptune.bulk_load` without any added value, I would consider just keeping the two separate.

Reply:
That's a good idea. And it would be consistent with how we do `copy_from_files` for Redshift.

Reply:
That's a good idea, I'll add `keep_files`.

I just looked at the Redshift module which has similar functionality. It has both a `copy` and `copy_from_files` function. The first one dumps the Parquet data to S3 and then calls `copy_from_files`. So in order to keep this consistent I think I should perhaps have `bulk_load` and `bulk_load_from_files`.

Writing to S3 doesn't seem like a lot but I think it allows us to ensure that the data gets written to S3 in the proper format. For example, the customer probably won't want `index=True`. On the other hand, this is the default value for `to_csv`. Having a function that just takes a DataFrame and writes it to S3 seems small, but it ensures that all the formatting and file structure is correct.

Reply:
Yeah I like the idea of `bulk_load` and `bulk_load_from_files`.  Sounds good!